### ISSUE TITLE
Add `require-schema-prefix` rule

### ISSUE THREAD

**Issue by keyvanm** [marcalexiei/eslint-plugin-zod#199]
### 🧩 Description

Similar to the `require-schema-suffix` rule, but for prefixes instead. Will enable patterns such as prefixing a `$` to zod schema names. 

### ⚙️ Options

Same as `require-schema-suffix`

### 🧾 Example Configuration

```ts
{
    rules: {
      "zod/require-schema-prefix": ["warn", { prefix: "$" }],  // $ should be the default IMO
    },
  }
```

### ✅ Valid Code Examples

```typescript
const $user = z.object({...})
const $url = z.url()
```

### ❌ Invalid Code Examples

```typescript
const user = z.object({...})
const url = z.url()
```

**Comment by marcalexiei** [comment]
New rules available to satisfy both prefix and suffix info: [consistent-schema-var-name](https://github.com/marcalexiei/eslint-plugin-zod/blob/main/docs/rules/consistent-schema-var-name.md).

[`require-schema-suffix` has been deprecated.](https://github.com/marcalexiei/eslint-plugin-zod/blob/main/docs/rules/require-schema-suffix.md#deprecation-notice)
